src/Event/Media/GetAlbumArt.php

```
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Event\Media;
+
+use App\Entity;
+use Symfony\Contracts\EventDispatcher\Event;
+
+class GetAlbumArt extends Event
+{
+    protected Entity\SongInterface $song;
+
+    protected ?string $albumArt = null;
+
+    public function __construct(Entity\SongInterface $song)
+    {
+        $this->song = $song;
+    }
+
+    public function getSong(): Entity\SongInterface
+    {
+        return $this->song;
+    }
+
+    public function setAlbumArt(?string $albumArt): void
+    {
+        $this->albumArt = !empty($albumArt)
+            ? $albumArt
+            : null;
+        $this->stopPropagation();
```

Stopping the propagation here means that after the first handler has called the `setAlbumArt` method no other handlers are called.

Could we explicitly check if `$albumArt` is `null` in order to let it propagate further if one API hasn't found the album art? I know this method is only called when the `!empty($albumArt)` in our handlers is passed so this wouldn't be a problem for us but I'd like to have this designed in a defensive manner just to be safe.